### DISCUSSION TITLE
export CSV human readable?

### DISCUSSION THREAD

**Discussion by guykatz** [microstream-one/microstream#570] [Q&A] [answered]
hi guys;
given the following simple object:
`
package com.example;
public class Model {
    String name;
    int age;
    public Model(String name, int age){
        this.name = name;
        this.age = age;
    }
}

package com.example;
public class Model2 {
    String name;
    int age;
    public Model(String job{
        this.name = job;
    }
}
`
storing as a root a list with Model and Model2 instances and exporting to CSV the resulting file are not really what I expected.
i though the output would be a nice human readable model and model 2 file with properties as columns and a raw per instance (one raw in my case)
however, the results seem to be scattered in a way that is hard to grasp. for example, in the string csv file we will have all string values from both instances (the job and the name)
so there is no real record per object here. 
in short, I am not sure the CSV is saved in the format most people would expect. maybe its more of a microstream csv format...
maybe there is a built in way to have the output CSV more human readable and orginized? :)
thanks

**Comment by hg-ms** [comment, marked as answer]
Hi,
You’re right the exported CSV is a bit Microstream specific.
The reason is that the source data is a java object graph that allows multiple references to another Object. E.g., Instances of Model and Model2 may reference the same String instance via the name property. To keep that information in the export the (unique) id of the referenced Object is exported instead of the value. Only primitive values like ‘int’, that can’t be referenced are included directly.
That way the original object graph can be reconstructed from the export.
At the moment there is no other export supplied. Although we discussed that already internally I can’t say if that will be implemented in the near future.
